node/block.go

```diff
@@ -45,6 +45,12 @@ func (node *Node) processBlock(ctx context.Context, pubSubMsg pubsub.Message) (e
 		return nil
 	}
 
+	// ignore messages from peers not yet tracked
+	if !node.PeerTracker.Has(from) {
+		log.Debugf("Ignoring new block from %s because we don't know this peer", from.Pretty())
```

Comment:
nit: should be "ignoring new block from unconnected peer %s" or similar- say something about not being connected to them. Also consider putting the block CID in here, might be helpful for debugging later.